lib/tools.js

```diff
@@ -1932,6 +1932,49 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Validates types of obj.common properties, if invalid types are used, an error is thrown.
+ * If attributes are not provided, no error is thrown
+ *
+ * @param {object} obj an object which will be validated
+ * @throws Error if a property has the wrong type
+ */
+function validateObjectCommonProperties(obj) {
+    if (!obj || !obj.common) return;
+
+    if (obj.common.name !== undefined && typeof obj.common.name !== 'string') {
+        throw new Error(`obj.common.name has wrong type (${typeof obj.common.name}) but has to be string`);
```

Comment:
Let's try to be consistent with our error messages. So like #704, I propose
```
obj.common.name has an invalid type! Expected "string", received "${typeof obj.common.name}"
```
Same goes for the other messages. Maybe after these two PRs are merged, we can even refactor those into a single method `throwWrongType(propertyName, receivedType, expectedType)`.